source/common/conn_pool/conn_pool_base.cc

```diff
@@ -107,6 +108,8 @@ bool ConnPoolImplBase::tryCreateNewConnection(float global_prefetch_ratio) {
     ASSERT(std::numeric_limits<uint64_t>::max() - connecting_stream_capacity_ >=
            client->effectiveConcurrentStreamLimit());
     ASSERT(client->real_host_description_);
+    // Increase the connecting capacity to reflect the streams this connection can serve.
+    state_.incrConnectingCapacity(client->effectiveConcurrentStreamLimit());
```

Comment:
Yeah, in practice these are set by config, and 

  google.protobuf.UInt32Value max_concurrent_streams = 2
      [(validate.rules).uint32 = {lte: 2147483647 gte: 1}];

so they're functionally capped at uint32.  Would you prefer I make all the base class types uint32_t, and the new types uint64 for overflow safety?

Reply:
ping?

Reply:
Sorry, I didn't realize that you had replied to comments until now.  I should have tagged the PR with waiting:any instead of waiting.

What if someone does not set config and ends up with the default concurrent_stream_limit of 0?
I did some tracing through the H2 implementation and found that initializeAndValidateOptions does translate a config value of 0 to 2147483647 as documented.  It seems to me that concurrent_stream_limit should never be 0 when it reaches this constructor.

Still, connecting_capacity_ is an uint32_t.  If there are more than 2 endpoints in the cluster, I think this will overflow when using the H2 default of 2**31-1 since I think these stats are per cluster, not per host/endpoint.